Dockerfile

```diff
@@ -46,107 +15,47 @@ RUN mkdir -p /home/unlock
 RUN chown -R node /home/unlock
 WORKDIR /home/unlock
 
-# copy packages info
-COPY --chown=node --from=manifests /opt/manifests .
-COPY --chown=node .prettierrc /home/unlock/.
-
-# yarn config
-COPY --chown=node .yarn/ /home/unlock/.yarn/
-COPY --chown=node .yarnrc.yml /home/unlock/.yarnrc.yml
-
-# add yarn cache folder to be used by docker buildkit
-RUN echo "cacheFolder: /home/unlock/yarn-cache" >> .yarnrc.yml
-
 # Setting user as root to handle apk install
 USER root
 
 # apk steps merged to leverage virtual install of package
 # allowing for removal after yarn dependencies install
-RUN apk add --no-cache --virtual .build-deps \
+RUN apt-get update \
+ && DEBIAN_FRONTEND=noninteractive \
+    apt-get install --no-install-recommends --assume-yes \
     bash \
     git \
-    openssh \
     python3 \
-    python3-dev \
-    py3-pip \
-    build-base \
-    openjdk11 \
-    && pip3 install --no-cache-dir virtualenv
+    build-essential \
+    postgresql \
+    default-jdk \
+    openjdk-11-jre
 
 # install deps
 USER node
 
-# attempt to create dir only for non-packages
-RUN if echo ${BUILD_DIR} | grep -q "packages" ; then echo "skipping"; else mkdir /home/unlock/${BUILD_DIR}; fi
+# copy all code
+COPY --chown=node . /home/unlock/
```

Comment:
doing this here means every PR will likely trigger anything below to be executed again because every PR contains file changes.
We need to copy only the package.json & yarn.lock first, then build the deps and finally copy the files so that dependency building can be cached.